crates/distribution-types/src/buildable.rs

```diff
@@ -71,6 +72,7 @@ impl<'a> SourceUrl<'a> {
             Self::Direct(dist) => dist.url,
             Self::Git(dist) => dist.url,
             Self::Path(dist) => dist.url,
+            Self::Directory(dist) => dist.url,
```

Comment:
I kind of wish the names were like... `Directory` and `Archive`? But `Archive` is a bit strange.

Reply:
I like `Archive`, pip uses that terminology too

Reply:
Is `Archive` in this context referring to a specific file like a `.tar.gz`? If that's always the case, then I like the name `Archive`.

Reply:
Yeah it needs to be a `.tar.gz` or `.zip` or `.tar.bz2` or a few other extensions that we support.

Reply:
I don't know that it's a perfect name because (e.g.) when you provide `Direct` (i.e., a direct URL), that has to be a direct URL to... an archive. So they're both archives?

Reply:
Honestly `Self::File` could be the right name.